storage/series.go

```diff
@@ -105,22 +105,58 @@ func (s *dbSeries) Tick() error {
 	// stale buckets, cheaply check this case first with a Rlock
 	s.RLock()
 	needsDrain := s.buffer.NeedsDrain()
+	needsBlockExpiry := s.needsBlockExpiry()
 	s.RUnlock()
 
-	if needsDrain {
+	if needsDrain || needsBlockExpiry {
 		s.Lock()
-		s.buffer.DrainAndReset()
+
+		if needsDrain {
+			s.buffer.DrainAndReset()
+		}
+
+		if needsBlockExpiry {
+			s.expireBlocks()
+		}
+
 		s.Unlock()
 	}
-
-	// TODO(r): expire and write blocks to disk, when done ensure this is async
-	// as ticking through series in a shard is done synchronously
 	return nil
 }
 
+// NB(xichen): this is called inside a RLock.
```

Comment:
Perhaps we should rename the methods to `WithRLock()` and `WithLock()` to make more readable and less necessary to add comments?